mitiq/pec/types/types.py

```diff
@@ -57,239 +56,62 @@ def __init__(
-            return
 
-        if channel_matrix.shape != (
-            self._dimension**2,
-            self._dimension**2,
+        elif channel_matrix.shape != (
+            dimension**2,
+            dimension**2,
         ):
             raise ValueError(
                 f"Arg `channel_matrix` has shape {channel_matrix.shape}"
                 " but the expected shape is"
-                f" {self._dimension ** 2, self._dimension ** 2}."
+                f" {dimension ** 2, dimension ** 2}."
             )
-        # TODO: Check if channel_matrix is a valid superoperator.
         self._channel_matrix = deepcopy(channel_matrix)
 
-    @staticmethod
-    def on_each(
-        circuit: cirq.CIRCUIT_LIKE,
-        qubits: Sequence[List[cirq.Qid]],
-        channel_matrix: Optional[npt.NDArray[np.complex64]] = None,
-    ) -> List["NoisyOperation"]:
-        """Returns a NoisyOperation(circuit, channel_matrix) on each
-        qubit in qubits.
-
-        Args:
-            circuit: A gate, operation, sequence of operations, or circuit.
-            channel_matrix: Superoperator representation of the noisy channel
-                which is generated when executing the input ``circuit`` on the
-                noisy quantum computer.
-            qubits: The qubits to implement ``circuit`` on.
-
-        Raises:
-            TypeError:
-                * If `qubits` is not iterable.
-                * If `qubits` is not an iterable of cirq.Qid's or
-                  a sequence of lists of cirq.Qid's of the same length.
-        """
-        try:
-            qubits = list(iter(qubits))
-        except TypeError:
-            raise TypeError("Argument `qubits` must be iterable.")
-
-        try:
-            num_qubits_needed = cirq.num_qubits(circuit)
-        except TypeError:
-            raise ValueError(
-                "Could not deduce number of qubits needed by `circuit`."
-            )
-
-        if all(isinstance(q, cirq.Qid) for q in qubits):
-            qubits = cast(Sequence[List[cirq.Qid]], [[q] for q in qubits])
-
-        if not all(len(qreg) == num_qubits_needed for qreg in qubits):
-            raise ValueError(
-                f"Number of qubits in each register should be"
-                f" {num_qubits_needed}."
-            )
-
-        noisy_ops = []  # type: List[NoisyOperation]
-        base_circuit = NoisyOperation.from_cirq(
-            circuit,
-            channel_matrix,
-        )._circuit
-        base_qubits = list(base_circuit.all_qubits())
-
-        for new_qubits in qubits:
-            try:
-                new_qubits = list(iter(new_qubits))
-            except TypeError:
-                new_qubits = list(new_qubits)
-
-            qubit_map = dict(zip(base_qubits, new_qubits))
-            new_circuit = base_circuit.transform_qubits(lambda q: qubit_map[q])
-
-            noisy_ops.append(NoisyOperation(new_circuit, channel_matrix))
-
-        return noisy_ops
-
-    def extend_to(
```

Comment:
May I ask why this was deleted? I do see that it was only "used "in tests- is it unlikely to be useful to Mitiq users?

Reply:
It doesn't seem to be needed in practice. Moreover it has some problems:

- It takes as input cirq qubits, so it's not multi-platform.
- It creates a mismatch between the `self._native_circuit` (not extended) and self.circuit (extended).
- I don't think `channel_matrix` is correctly extended in a consistent way.

We could fix these problems, but is it worth the effort and the risk of creating subtle bugs?

Reply:
Makes sense, agree to remove.